# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1483721533 18000
#      Fri Jan 06 11:52:13 2017 -0500
# Node ID b8cef5f05dcd3e30567bb1488e7f00b7e66cc2de
# Parent  06c248729e6ff5e3079bb9d9152084c331576109
use OCTAVE_CHECK_LIB to check for PCRE library (bug #49993)

* configure.ac: Use OCTAVE_CHECK_LIB to check for PCRE library and
header files.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1334,25 +1334,19 @@ OCTAVE_CHECK_LIB(qhull, QHull,
   [warn_qhull=
   OCTAVE_CHECK_QHULL_VERSION
   OCTAVE_CHECK_LIB_QHULL_OK(
     [AC_DEFINE(HAVE_QHULL, 1, [Define to 1 if Qhull is available.])],
     [warn_qhull="Qhull library found, but does not seem to work properly.  This will result in loss of functionality of some geometry functions.  Please try recompiling the library with -fno-strict-aliasing."])])
 
 ### Check for PCRE regex library.
 
-PKG_CHECK_EXISTS([libpcre],
-  [AC_CHECK_HEADERS([pcre.h pcre/pcre.h])
-   PCRE_CPPFLAGS=`$PKG_CONFIG --cflags-only-I libpcre`
-   PCRE_LDFLAGS=`$PKG_CONFIG --libs-only-L libpcre`
-   PCRE_LIBS=`$PKG_CONFIG --libs-only-l libpcre`
-   AC_SUBST(PCRE_CPPFLAGS)
-   AC_SUBST(PCRE_LDFLAGS)
-   AC_SUBST(PCRE_LIBS)],
-  [AC_MSG_ERROR([to build Octave, you must have the PCRE library and header files installed])])
+OCTAVE_CHECK_LIB(pcre, PCRE,
+  [], [pcre.h pcre/pcre.h], [pcre_compile], [], [], [],
+  [libpcre], [REQUIRED])
 
 ### Check for ZLIB library.
 
 OCTAVE_CHECK_LIB(z, ZLIB,
   [ZLIB library not found.  Octave will not be able to save or load compressed data files or HDF5 files.],
   [zlib.h], [gzclearerr])
 
 ### Also define HAVE_ZLIB if libz is found.
